lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = function initialize(_agent, elastic, _moduleName, shim) {
+  shim.setDatastore(shim.ELASTICSEARCH)
+  shim.setParser(queryParser)
+
+  shim.recordQuery(elastic.Transport.prototype, 'request', function wrapQuery(shim, _, __, args) {
+    const ctx = this
+    return {
+      query: JSON.stringify(args?.[0]),
+      promise: true,
+      opaque: true,
+      inContext: function inContext() {
+        getConnection.call(ctx, shim)
+      }
+    }
+  })
+}
+
+function queryParser(params) {
+  params = JSON.parse(params)
+  const path = params.path.split('/')
+  let operation = params.method
+  let collection = path?.[1] || 'other'
+
+  // if there's no specified index, the second path element is the operation
+  if (collection[0] === '_') {
+    operation = collection
+    collection = 'other' // presumably search of all indices
+  }
+
+  // the substance of the query may be in querystring or in body.
+  let queryParam = {}
+  if (typeof params.querystring === 'object' && Object.keys(params.querystring) > 0) {
+    queryParam = params.querystring
+  } else if (typeof params.body === 'object' && Object.keys(params.body).length > 0) {
```

Comment:
looks like body could also be bulkBody

Reply:
I also think we should favor body/bulkBody over querystring as in most cases the body is the "query"

Reply:
Somewhere in the Elastic docs, I saw a notation about precedence when both request body and querystring are supplied. I *think* request body gets ignored if there's a string. (I'll verify.)

Reply:
One of our test cases includes both a querystring and a bulkBody. In this case, the bulkBody has much more actionable information than the `{refresh: true)` of the querystring, so I'd vote to prioritize the body or requestBody.